### PR TITLE
BREAKING: DM Slash Commands, new Interactions API changes

### PR DESCRIPTION
**Pull request by DjDeveloperr**
- Use more concise names for Interaction typings. (BREAKING)
- Update with the typings from docs - we had some old ones. (BREAKING)
- Fixed `allowed_mentions` object.
- Added `CommandsLoader` for easy command loading from files/directory.
- Use Big Int for Bit Fields. (BREAKING)
- Remove redundant `awaitSync` util.
- Add `USE_SLASH_COMMANDS` permission bit field.
- Update typings (and code) with new changes to Interactions API: https://github.com/discord/discord-api-docs/pull/2615. (BREAKING).

Interaction API changes include (I might have forgotten some):
- Updated Response Types.
- Added `Interaction#reply`, `Interaction#defer`, `Interaction#resolved`.
- With `resolved`, arguments with types `USER`, `ROLE` or `CHANNEL` now resolve to actual objects.
- Due to partial nature of `CHANNEL` object in `resolved`, I've added `InteractionChannel` with a `resolve` method that resolves to full object from cache if it is there.
- I've added `InteractionUser` which will also contain optional `member` property as Union types make it harder to work with options (arguments).

~~Waiting for above mentioned PR to merge, DM Slash Commands to be out on stable.~~

### CODE REVIEW

File: src/types/slash.ts
Comment (Helloyunho) [inline src/types/slash.ts:28]: I think moving this to another file will be better since it might be useful in other things too

Comment (DjDeveloperr) [reply src/types/slash.ts]: Done.

**Review by Helloyunho — APPROVED** [review]
lgtm!

**Comment by DjDeveloperr** [comment]
Don't merge until DM Slash Commands are out on Stable.

**Comment by Helloyunho** [comment]
Some are split into #107 and #108

**Comment by DjDeveloperr** [comment]
Changes have landed. This PR needs more testing before being landed, but you can review now.

**Review by Helloyunho — APPROVED** [review]

File: src/models/slashClient.ts
Comment (Helloyunho) [inline src/models/slashClient.ts:380]: maybe we can move this to utils/index.ts

**Review by Helloyunho — CHANGES_REQUESTED** [review]

File: src/structures/slash.ts
Comment (DjDeveloperr) [reply src/structures/slash.ts:257]: Embed is assignable to EmbedPayload 🤔 I used EmbedPayload so we can put raw embed json instead of having to call Embed constructor

Comment (Helloyunho) [reply src/structures/slash.ts:257]: does it convert without doing anything?

Comment (DjDeveloperr) [reply src/structures/slash.ts:257]: Why would we need to convert it?

Comment (Helloyunho) [reply src/structures/slash.ts:257]: because EmbedPayload and Embed has different properties

**Review by Helloyunho — APPROVED** [review]
lgtm!

**Comment by Helloyunho** [comment]
@arija-ari you can probably use `@slash("name", "guildID")` to limit the command to the specific guild only
you can find the definition here https://github.com/harmonyland/harmony/blob/b37bf5dcbed321fd1b1b06595e2a4f94be3c8123/src/interactions/decorators.ts#L187-L190

**Pull request merged**
